agent/lib/kontena/launchers/cadvisor.rb

```diff
@@ -67,14 +69,17 @@ def create_container(image)
           '--listen_ip=127.0.0.1',
           '--port=8989',
           '--storage_duration=2m',
-          '--housekeeping_interval=30s'
+          '--housekeeping_interval=10s',
+          '--disable_metrics=tcp,disk'
         ],
+        'Labels' => {
+          'io.kontena.agent.version' => Kontena::Agent::VERSION
+        },
         'Volumes' => volume_mappings,
         'HostConfig' => {
           'Binds' => volume_binds,
           'NetworkMode' => 'host',
-          'PidMode' => 'host',
-          'Privileged' => true,
+          'CpuShares' => 128,
```

Comment:
I'd say 256M should allow some temporary mem hogging also and not get immediate OOM shot-in-the-head.